future-apps/payroll/contracts/Payroll.sol

```diff
@@ -541,15 +552,26 @@ contract Payroll is EtherTokenConstant, IsContract, AragonApp { //, IForwarder {
             toDate = timestamp;
         }
 
-        // Compute owed amount
-        uint256 owed = employee.accruedValue.add(_getOwedSalary(_employeeId, toDate));
-        if (owed == 0) {
+        // Compute owed amount, set to max int in case of overflow
+        uint256 owed = employee.accruedValue + _getOwedSalary(_employeeId, toDate);
+        if (owed < employee.accruedValue) {
+            owed = MAX_UINT256;
+        }
+        if (owed == 0 || owed < _amount) {
             return false;
         }
 
         // Update last payroll date and accrued value first thing (to avoid re-entrancy)
         employee.lastPayroll = timestamp;
-        employee.accruedValue = 0;
+        uint256 toPay;
+        if (_amount > 0 && owed > _amount) {
+            // no need for safemath as here we know owed >= _amount
+            employee.accruedValue = owed - _amount;
+            toPay = _amount;
+        } else {
+            employee.accruedValue = 0;
+            toPay = owed;
+        }
```

Comment:
I think we should avoid removing employees if a partial amounts was requested, right? (see https://github.com/protofire/aragon-apps/pull/113/files#diff-9d7adaffbcce19867f027a983a30bcc7R602)

Reply:
Yes, but if partial amount was requested this won't hold, right? `employee.accruedValue == 0`